gitmux: inti at 0.7.10

nialov opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Add `gitmux` which is a `go` app made for the simple purpose of printing `git status` prettily in `tmux` statusbar.

It is distributed as a single binary and therefore I've not added this as a `tmuxPlugin` as you could in theory use it for other purposes as `tmux` is not required (prints to stdout which can be captured by `tmux`). As I'm writing this request I'm wondering if `git` should be added as a runtime dependency as that is what it uses to provide its main functionality (get `git status` in current directory). It can create default config files and such for itself without `git` however and doesn't error without it either.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

nialov commented:
Fixed whitespace in string block with the force push.

FliegendeWurst commented on pkgs/applications/misc/gitmux/default.nix:
```diff
@@ -0,0 +1,38 @@
+  # but it requires networking as it git clones a repo from github
+  doCheck = false;
+
+  # Added check that binary can print --help message without error
+  doInstallCheck = true;
+  installCheckPhase = ''
+    runHook preInstallCheck
+
+    $out/bin/gitmux --help > /dev/null
+
+    runHook postInstallCheck
+  '';
```
```suggestion
  prePatch = ''
    substituteInPlace gitmux.go --replace '<<development version>>' "${version}"
  '';

  passthru.tests.version =
    testVersion {
      package = gitmux;
      command = "gitmux -V";
    };
```

nialov replied:
Thank you, added the suggested changes.

FliegendeWurst reviewed (commented):
Works as expected

nialov commented:
Another pull request of mine was merged a few days ago, so I'm already in the maintainer list. Should I remove that change from this request or does it matter?

FliegendeWurst commented:
> Another pull request of mine was merged a few days ago, so I'm already in the maintainer list. Should I remove that change from this request or does it matter?

You should remove it (see the failed checks)

FliegendeWurst commented:
I think it would be best to squash the commits.

qbit commented on pkgs/applications/misc/gitmux/default.nix:
```diff
@@ -0,0 +1,38 @@
+  doCheck = false;
+
+  prePatch = ''
+    substituteInPlace gitmux.go --replace '<<development version>>' "${version}"
```
This `prePatch` should be replaced with a `ldflags = [ "-X main.version=${version}" ];` line.

nialov replied:
Thanks for review, change implemented.

nialov commented:
Commits have been squashed!

qbit commented:
LGTM!

qbit commented:
Also tested on aarch64-darwin without issuee!

onny commented:
Fails to build for me
```
error: Function called without required argument "testVersion" at /home/onny/.cache/nixpkgs-review/pr-154304/nixpkgs/pkgs/applications/misc/gitmux/default.nix:1
```

nialov commented:
> Fails to build for me
> 
> ```
> error: Function called without required argument "testVersion" at /home/onny/.cache/nixpkgs-review/pr-154304/nixpkgs/pkgs/applications/misc/gitmux/default.nix:1
> ```

Is there a change in ``callPackage`` in relation to ``testVersion``? Anyone know?

nialov committed:
gitmux: init at 0.7.10

nialov commented:
Sorry for the delay. Changes:
-  Changed from `testVersion` input to `testers.testVersion`. Should fix error reported by onny.
-  Rebased onto current `master`.
-  Updated package version to `0.7.10` (latest release).
-  Changed parent directory for the package to `pkgs/applications/version-management/git-and-tools`. 
 
Hopefully the changes are appropriate. No radical change in functionality as far as I am aware.

onny commented:
Result of `nixpkgs-review pr 154304` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gitmux</li>
  </ul>
</details>

onny commented:
Thank you for fixing the issues and contributing the app :)
